### ISSUE TITLE
AttributeError: module 'keyboard._darwinkeyboard' has no attribute 'type_unicode'

### ISSUE THREAD

**Issue by ftpersonal** [boppreh/keyboard#89]
Hello, 
          I am testing your "write.py" example on OSX Sierra 10.12.6. I tested this using the native python 2.7.10 and brew install of python 3 at 3.6.2. Used pip to install keyboard once in both python environments. Both had identical errors listed below. 

The program actually works, but then crashes with the following error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/keyboard/__init__.py", line 570, in write
    scan_code, modifiers = _os_keyboard.map_char(letter)
  File "/usr/local/lib/python3.6/site-packages/keyboard/_darwinkeyboard.py", line 390, in map_char
    return key_controller.map_char(character)
  File "/usr/local/lib/python3.6/site-packages/keyboard/_darwinkeyboard.py", line 305, in map_char
    return self.key_map.character_to_vk(character)
  File "/usr/local/lib/python3.6/site-packages/keyboard/_darwinkeyboard.py", line 153, in character_to_vk
    raise ValueError("Unrecognized character: {}".format(character))
ValueError: Unrecognized character: enter

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "write.py", line 14, in <module>
    keyboard.write(line)
  File "/usr/local/lib/python3.6/site-packages/keyboard/__init__.py", line 584, in write
    _os_keyboard.type_unicode(letter)
AttributeError: module 'keyboard._darwinkeyboard' has no attribute 'type_unicode'
```

I cannot tell if this is a configuration, dependency or other problem. But it is an issue with one of the standard examples. 

Thanks for your Open Source contributions.

Regards,
-FT

**Comment by boppreh** [comment]
Hi @ftpersonal

I'm sorry, it seems to be a problem in the library itself. Unfortunately OS X support is pretty experimental at the moment, since I don't have a mac and depend on external contributions for this bit.

There seems to be two problems: first 'enter' is not being mapped to any key, and second that the OS X backend hasn't implemented the necessary functions for typing arbitrary keys (`type_unicode`). I've attempted to fix the first issue (https://github.com/boppreh/keyboard/commit/f740496780ea7f1e3ebdf422db99a328b5bbea34) but I need your help testing to make sure it works.

Can you try the latest version from the master branch? Hopefully it will be able to run the example code.

**Comment by gerardo** [comment]
Hi @boppreh,

I was facing this problem today. Just confirming that your fix works for me 👍

**Comment by antontarasenko** [comment]
@boppreh 

Hi! Another happy user of your package has this error.

I'm doing `keyboard.add_abbreviation('email', 'my.long.email@example.com')` like this:
```
import sys
sys.path.insert(0, '__PATH_TO_LATEST_KEYBOARD_VERSION__')
import keyboard

if __name__ == '__main__':
    keyboard.add_abbreviation('email', 'my.long.email@example.com')
    keyboard.wait()
```

Then I run this file, type `email<space>` in a terminal or text editor, and the script returns:
```
...
ValueError: Unrecognized character: backspace
...
AttributeError: module 'keyboard._darwinkeyboard' has no attribute 'type_unicode'
```

I'm using the fixed version as in https://github.com/boppreh/keyboard/commit/f740496780ea7f1e3ebdf422db99a328b5bbea34

**Comment by boppreh** [comment]
Hi @antontarasenko 

I never had a Mac, so imagine my surprise when I learned that the "backspace" key in a Mac is named "delete", and the "delete" is "forward delete". The library seems to be going by that, so the "backspace" key was missing.

I changed the names around to the more cross-platform way, which I understand may break some things, but is a choice that must be named. Any opinion on this, @glitchassassin ?

I also added a bunch more key codes. Again, I can't test any of this, so @antontarasenko , could you test the latest master version, ccfdc1c?

**Comment by glitchassassin** [comment]
"Backspace" can easily be mapped to "delete," since there's no semantic overlap on Mac. Mapping "delete" to "forward delete" is the controversial bit.

I tend to think that "delete" should correspond to the actual "delete" key on Mac; preserving the intuitive (and correct) key names for Mac development seems more useful and friendly than creating a cross-platform mapping that breaks those names.

I'd suggest mapping cross-platform synonyms in the character_to_vk method (as in the PR I updated) rather than just replacing the canonical OS X names. That allows for multiple valid names (as in "command" vs. "windows" or "backspace" vs "delete").

I'll try to put in some testing soon, but it'll probably be late this week or the week after.

Thanks!

**Comment by boppreh** [comment]
Sounds good.

**Comment by antontarasenko** [comment]
@boppreh 

The new version works perfectly. That was fast. Thank you!

**Comment by antontarasenko** [comment]
@boppreh 

A new chapter in this Mac keyboard story. When I do `keyboard.wait('esc')` (or with `'escape'`) and press ESC, it says `ValueError: Unrecognized character: alt gr`.

**Comment by DZGoldman** [comment]
I have the latest version, and I still get 
```
module 'keyboard._darwinkeyboard' has no attribute 'type_unicode'
```

from `keyboard.add_abbreviation('x', 'xxx')`

Using a mac OSX v. 10.11.6

**Comment by NoSkillGuy** [comment]
I am Mac OS Sierra, I have the same issue.

**Comment by boppreh** [comment]
Hi @NoSkillGuy 

Are you sure you are using the latest version, v0.13.1? I just downloaded that version from PyPI and there's `def type_unicode` in `_darwinkeyboard.py`. It's not impossible something is being mangled, but please do a

    pip install --upgrade keyboard

to double check.

About alt gr, there's an open issue about that: #117. I have a fix in the master branch (mapping "alt gr" as simple "alt"), but I'm looking for feedback if this is surprising or if it's ok.

**Comment by speggos** [comment]
Also getting the following error:

AttributeError: module 'keyboard._darwinkeyboard' has no attribute 'map_name'

I am using version 0.13.1. Have others been able to get around this issue?

**Comment by boppreh** [comment]
@speggos The `map_name` issue was fixed on master, and the fix is now available on version 0.13.2 on PyPI.
